File: packages/go/sentinelops.go
package sentinelops

import (
	"bytes"
	"encoding/json"
	"net/http"
	"time"
)

type Config struct {
	Endpoint    string
	APIKey      string
	TenantID    string
	ServiceName string
}

type Incident struct {
	Title    string \json:"title"\
	Division string \json:"division"\
	Severity string \json:"severity"\
	Details  string \json:"details,omitempty"\
}

type Client struct {
	cfg Config
	hc  *http.Client
}

func NewClient(cfg Config) *Client {
	return &Client{
		cfg: cfg,
		hc:  &http.Client{Timeout: 5 * time.Second},
	}
}

func (c *Client) ReportIncident(inc Incident) error {
	body, _ := json.Marshal(inc)
	req, err := http.NewRequest("POST", c.cfg.Endpoint+"/api/v1/incidents", bytes.NewBuffer(body))
	if err != nil {
		return err
	}
	req.Header.Set("Content-Type", "application/json")
	req.Header.Set("Authorization", "Bearer "+c.cfg.APIKey)
	req.Header.Set("X-Sentinel-Tenant", c.cfg.TenantID)
	_, err = c.hc.Do(req)
	return err
}

File: packages/python/sentinelops.py
import requests
from datetime import datetime

class SentinelClient:
    def __init__(self, endpoint: str, api_key: str, tenant_id: str, service_name: str = 'python-service'):
        self.endpoint = endpoint.rstrip('/')
        self.api_key = api_key
        self.tenant_id = tenant_id
        self.service_name = service_name

    def report_incident(self, title: str, division: str, severity: str = 'HIGH', details: str = ''):
        payload = {
            'title': title,
            'division': division,
            'severity': severity,
            'details': details,
            'service': self.service_name,
            'timestamp': datetime.utcnow().isoformat()
        }
        headers = {
            'Content-Type': 'application/json',
            'Authorization': f'Bearer {self.api_key}',
            'X-Sentinel-Tenant': self.tenant_id
        }
        try:
            r = requests.post(f'{self.endpoint}/api/v1/incidents', json=payload, headers=headers, timeout=5)
            return r.json()
        except Exception as e:
            return {'success': False, 'error': str(e)}
